src/OpenFeature/Providers/Memory/InMemoryProvider.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            var changed = this._flags.Keys.ToList();
+            if (flags == null)
+            {
+                this._flags = new Dictionary<string, Flag>();
+            }
+            else
+            {
+                this._flags = new Dictionary<string, Flag>(flags); // shallow copy
+            }
+            changed.AddRange(this._flags.Keys.ToList());
+            var @event = new ProviderEventPayload
+            {
+                Type = ProviderEventTypes.ProviderConfigurationChanged,
+                ProviderName = _metadata.Name,
+                FlagsChanged = changed, // emit all
+                Message = "flags changed",
+            };
+            await this.EventChannel.Writer.WriteAsync(@event).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<bool>> ResolveBooleanValue(
+            string flagKey,
+            bool defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(
+            string flagKey,
+            string defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(
+            string flagKey,
+            int defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(
+            string flagKey,
+            double defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<Value>> ResolveStructureValue(
+            string flagKey,
+            Value defaultValue,
+            EvaluationContext? context = null)
+        {
+            return Task.FromResult(Resolve(flagKey, defaultValue, context));
+        }
+
+        private ResolutionDetails<T> Resolve<T>(string flagKey, T defaultValue, EvaluationContext? context)
+        {
+            if (!this._flags.TryGetValue(flagKey, out var flag))
+            {
+                throw new FlagNotFoundException($"flag {flag} not found");
+            }
+            else
+            {
+                if (typeof(Flag<T>).Equals(flag.GetType()))
+                {
+                    return ((Flag<T>)flag).Evaluate(flagKey, defaultValue, context);
+                }
+                else
+                {
+                    throw new TypeMismatchException($"flag {flag} not found");
```

Comment:
This looks to be the wrong message, copy paste from the `FlagNotFoundException`.

Reply:
Nice catch.